### DISCUSSION TITLE
Issue getting styles to show with render function

### DISCUSSION THREAD

**Discussion by ryanbeaulieu** [resend/react-email#1663] [Q&A]
I am attempting to use some components from Material UI. When I create a react-email template using Material UI and run the react-email in the browser, the styles all render properly and if I use the built in "Send" feature in the UI that uses Resend, everything works as expected.

However, if I use the same exact email template and run it through the `render` function, the resulting HTML text does not contain any of the styles from Material UI. Not sure if anyone has tried using MUI and has run into this issue. Any help would be appreciated.

**Comment by ryanbeaulieu** [comment]
Here is what the markup looks like when inspecting the iframe when I run react-email 
`<!DOCTYPE html PUBLIC &quot;-//W3C//DTD XHTML 1.0 Transitional//EN&quot; &quot;http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd&quot;>
<html dir=&quot;ltr&quot; lang=&quot;en&quot;>

  <head>
    <meta content=&quot;text/html; charset=UTF-8&quot; http-equiv=&quot;Content-Type&quot; />
    <meta name=&quot;x-apple-disable-message-reformatting&quot; />
  </head>

  <body>
    <style data-emotion=&quot;css 1hxq67s&quot;>
      .css-1hxq67s {
        font-family: &quot;Roboto&quot;, &quot;Helvetica&quot;, &quot;Arial&quot;, sans-serif;
        font-weight: 500;
        font-size: 0.875rem;
        line-height: 1.75;
        letter-spacing: 0.02857em;
        text-transform: uppercase;
        min-width: 64px;
        padding: 6px 16px;
        border-radius: 4px;
        -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
        transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
        color: #fff;
        background-color: #1976d2;
        box-shadow: 0px 3px 1px -2px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 1px 5px 0px rgba(0, 0, 0, 0.12);
      }

      .css-1hxq67s:hover {
        -webkit-text-decoration: none;
        text-decoration: none;
        background-color: #1565c0;
        box-shadow: 0px 2px 4px -1px rgba(0, 0, 0, 0.2), 0px 4px 5px 0px rgba(0, 0, 0, 0.14), 0px 1px 10px 0px rgba(0, 0, 0, 0.12);
      }

      @media (hover: none) {
        .css-1hxq67s:hover {
          background-color: #1976d2;
        }
      }

      .css-1hxq67s:active {
        box-shadow: 0px 5px 5px -3px rgba(0, 0, 0, 0.2), 0px 8px 10px 1px rgba(0, 0, 0, 0.14), 0px 3px 14px 2px rgba(0, 0, 0, 0.12);
      }

      .css-1hxq67s.Mui-focusVisible {
        box-shadow: 0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12);
      }

      .css-1hxq67s.Mui-disabled {
        color: rgba(0, 0, 0, 0.26);
        box-shadow: none;
        background-color: rgba(0, 0, 0, 0.12);
      }
    </style>
    <style data-emotion=&quot;css 1hw9j7s&quot;>
      .css-1hw9j7s {
        display: -webkit-inline-box;
        display: -webkit-inline-flex;
        display: -ms-inline-flexbox;
        display: inline-flex;
        -webkit-align-items: center;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
        -webkit-box-pack: center;
        -ms-flex-pack: center;
        -webkit-justify-content: center;
        justify-content: center;
        position: relative;
        box-sizing: border-box;
        -webkit-tap-highlight-color: transparent;
        background-color: transparent;
        outline: 0;
        border: 0;
        margin: 0;
        border-radius: 0;
        padding: 0;
        cursor: pointer;
        -webkit-user-select: none;
        -moz-user-select: none;
        -ms-user-select: none;
        user-select: none;
        vertical-align: middle;
        -moz-appearance: none;
        -webkit-appearance: none;
        -webkit-text-decoration: none;
        text-decoration: none;
        color: inherit;
        font-family: &quot;Roboto&quot;, &quot;Helvetica&quot;, &quot;Arial&quot;, sans-serif;
        font-weight: 500;
        font-size: 0.875rem;
        line-height: 1.75;
        letter-spacing: 0.02857em;
        text-transform: uppercase;
        min-width: 64px;
        padding: 6px 16px;
        border-radius: 4px;
        -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
        transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
        color: #fff;
        background-color: #1976d2;
        box-shadow: 0px 3px 1px -2px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 1px 5px 0px rgba(0, 0, 0, 0.12);
      }

      .css-1hw9j7s::-moz-focus-inner {
        border-style: none;
      }

      .css-1hw9j7s.Mui-disabled {
        pointer-events: none;
        cursor: default;
      }

      @media print {
        .css-1hw9j7s {
          -webkit-print-color-adjust: exact;
          color-adjust: exact;
        }
      }

      .css-1hw9j7s:hover {
        -webkit-text-decoration: none;
        text-decoration: none;
        background-color: #1565c0;
        box-shadow: 0px 2px 4px -1px rgba(0, 0, 0, 0.2), 0px 4px 5px 0px rgba(0, 0, 0, 0.14), 0px 1px 10px 0px rgba(0, 0, 0, 0.12);
      }

      @media (hover: none) {
        .css-1hw9j7s:hover {
          background-color: #1976d2;
        }
      }

      .css-1hw9j7s:active {
        box-shadow: 0px 5px 5px -3px rgba(0, 0, 0, 0.2), 0px 8px 10px 1px rgba(0, 0, 0, 0.14), 0px 3px 14px 2px rgba(0, 0, 0, 0.12);
      }

      .css-1hw9j7s.Mui-focusVisible {
        box-shadow: 0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12);
      }

      .css-1hw9j7s.Mui-disabled {
        color: rgba(0, 0, 0, 0.26);
        box-shadow: none;
        background-color: rgba(0, 0, 0, 0.12);
      }
    </style><button class=&quot;MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1hw9j7s&quot; tabindex=&quot;0&quot; type=&quot;button&quot;>Testing</button>
  </body>

</html>`



Here is the markup if I run the render funciton in my react app 
```
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><!--$-->
<html dir="ltr" lang="en">

  <head>
    <meta content="text/html; charset=UTF-8" http-equiv="Content-Type" />
    <meta name="x-apple-disable-message-reformatting" />
  </head>

  <body><button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root" tabindex="0" type="button">Testing</button></body>

</html><!--/$-->
```

**Comment by gabrielmfern** [comment]
First thing I would say is don't use Material UI, unless they make components specific to React Email. React Email does not handle imported styles, which might be the cause of the problem here, and Material UI is meant for browser HTML not email HTML which means the support for email clients is going to be terrible.
React Email's components improve as much as they can for you to do the basics on them without issues, but email client support is still something to keep in mind when doing your own styles.

That being said, I can't vouch for the support we would have if used with something like MUI.
